editor/src/components/canvas/controls/classname-select.tsx

```diff
@@ -2,26 +2,135 @@
 
 import React from 'react'
 import { jsx } from '@emotion/react'
+import styled from '@emotion/styled'
 
 import TailWindList from '../../../core/third-party/tailwind-all-classnames.json'
-import WindowedSelect from 'react-windowed-select'
+import WindowedSelect, {
+  components,
+  createFilter,
+  IndicatorProps,
+  MultiValueProps,
+  ValueContainerProps,
+} from 'react-windowed-select'
+import type { StylesConfig } from 'react-select'
+import type { Option } from 'react-select/src/filters'
+import chroma from 'chroma-js'
 
 import * as EditorActions from '../../editor/actions/action-creators'
 import { betterReactMemo } from '../../../uuiui-deps'
-import { useColorTheme, UtopiaTheme } from '../../../uuiui'
+import { useColorTheme } from '../../../uuiui'
 import { useEditorState } from '../../editor/store/store-hook'
 import { MetadataUtils } from '../../../core/model/element-metadata-utils'
 import * as EP from '../../../core/shared/element-path'
 import * as PP from '../../../core/shared/property-path'
 import { jsxAttributeValue } from '../../../core/shared/element-template'
 import { emptyComments } from '../../../core/workers/parser-printer/parser-printer-comments'
 
+const ChromaBlack = chroma('black')
+
+interface TailWindOption {
+  label: string
+  value: string
+  categories?: string[]
+}
+
 const TailWindOptions = TailWindList.classNames.map((className, index) => ({
   label: className,
   value: className,
 }))
 
+const DropdownIndicator = betterReactMemo(
+  'DropdownIndicator',
+  (props: IndicatorProps<TailWindOption, true>) => (
+    <components.DropdownIndicator {...props}>
+      <span> ↓ </span>
+    </components.DropdownIndicator>
+  ),
+)
+
+const ClearIndicator = () => null
+const IndicatorSeparator = () => null
+
+const NoOptionsMessage = betterReactMemo('NoOptionsMessage', (props: any) => (
+  <span {...props}>No results found</span>
+))
+
+const AngledStripe = styled.div({
+  width: 5,
+  height: 22,
+  borderRadius: 0,
+  transform: 'skewX(-11deg)',
+})
+
+const getColorForCategory = (category: string) => {
+  if (category === 'layout') {
+    return '#5FACFF'
+  } else if (category === 'typography') {
+    return '#F7B500'
+  } else if (category === 'interaction') {
+    return '#B620E0'
+  } else return 'pink'
+}
+
+const MultiValueContainer = betterReactMemo(
+  'MultiValueContainer',
+  (props: MultiValueProps<TailWindOption>) => {
+    const { data } = props
+    const stripes: jsx.JSX.Element[] = React.useMemo(() => {
+      const categories = (data?.categories as Array<string>) ?? []
+      if (categories.length > 0) {
+        return categories.map((category, index) => (
+          <AngledStripe
+            key={data?.label ?? index}
+            style={{ backgroundColor: getColorForCategory(category) }}
+          />
+        ))
+      } else {
+        return []
+      }
+    }, [data])
```

Comment:
is `data` here referentially stable? or would it be better to feed it `label` and `categories`? (genuine question, I can't tell from just reading the code)

Reply:
fixed